server/app/src/androidTest/java/sh/calaba/instrumentationbackend/query/Query.java

```diff
@@ -104,6 +104,8 @@ else if (o instanceof Map) {
 
 	@SuppressWarnings("unchecked")
 	public static List<UIQueryAST> parseQuery(String query) {
+		// Replace '\\' with '\\\\' to prevent NoViableAltException ANTLR exception due to incorrect parsing
```

Comment:
Are you saying this is a workaround to a bug in antlr? - or is it a quirk/shortcoming of our grammar? I read it as a bug in antlr - if that's true please add link to the bug report in the comment.

Reply:
No. ANTLR logic is correct, because it expects valid expression. `\\` is not valid expression, but it's valid text value. I can't say that I like this solution, but it's working and safety.
For this issue we have 2 solutions: escape double backslashes only or change parser logic which seems to be dangerous, because it can break valid queries  https://groups.google.com/forum/#!topic/antlr-discussion/eiXpuz68zsE

Reply:
Important info: 
`\` in labels/inputs/buttons is a valid value on Android. In UI you can see backslash. Android escapes it implicitly.
Result for element with one backslash:
```
"results": [
        {
            "rect": {
                "x": 0,
                "width": 1080,
                "y": 360,
                "center_y": 432,
                "center_x": 540,
                "height": 144
            },
            "visible": true,
            "contentDescription": null,
            "description": "android.widget.CheckedTextView{8e382e V.ED..... ......I. 0,0-1080,144 #1020014 android:id/text1}",
            "id": "text1",
            "tag": null,
            "text": "D\\A",
            "class": "android.widget.CheckedTextView",
            "enabled": true
        }
```

Reply:
Aha, I see now. If you explain that in the code comment I think future generations of hackers will be thankful :-)

Reply:
Yep. I will add more info into comment.

Reply:
Added